### PR TITLE
Implemented search test with pageobject

### PR DESCRIPTION
**Pull request by andkochergin**

### CODE REVIEW

File: spec/features/search_spec.rb [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+
 require 'spec_helper'
 
 RSpec.describe 'Onliner search' do
+  let(:home_page) { ::PageObjects::Pages::HomePage.new }
+
   it 'is iPhone X released in 2017' do
     visit '/'
-    find(:xpath, '//input[@class="fast-search__input"]').set "iPhone X"
+    find(:xpath, '//input[@class="fast-search__input"]').set 'iPhone X'
```
Comment (yahorzhylinski) [inline spec/features/search_spec.rb]: What about to wrap this code to search form?

File: spec/features/search_spec.rb [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
     within_frame find('iframe.modal-iframe') do
-      first(:xpath,'//a[contains(text(),"iPhone X")]').click
+      first(:xpath, '//a[contains(text(),"iPhone X")]').click
```
Comment (yahorzhylinski) [inline spec/features/search_spec.rb]: What about to wrap this code to search form?

File: spec/features/search_spec.rb [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
     end
     expect(self).to have_xpath('//table[@class="product-specs__table"] '\
                                '//span[contains(text(),"2017")]')
     expect(self).to have_xpath('//td[contains(text(),"Дата выхода")] '\
                                '/following-sibling::td[1]/span')
   end
+
+  it 'search shows right products for queries' do
```
Comment (yahorzhylinski) [inline spec/features/search_spec.rb]: I think word `search` here is not necessary

File: spec/features/search_spec.rb [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+    visit '/'
+    home_page.search_queries.each do |search_query|
+      home_page.type_text_to_search_field(search_query[:query])
+      expect(search_query[:query]).to
```
Comment (yahorzhylinski) [inline spec/features/search_spec.rb]: Why you moved eq(...) to the separate line?

File: spec/page_objects/pages/home_page.rb [hunk 5]
```diff
@@ -17,6 +17,47 @@ def have_all_news_blocks_loaded?
         expect(all('.b-main-page-blocks-header-2').size).to eq(news_blocks.size)
       end
 
+      def type_text_to_search_field(query)
```
Comment (yahorzhylinski) [inline spec/page_objects/pages/home_page.rb]: What about to wrap this code to search form?

File: spec/features/search_spec.rb (hunk 1 above)
Comment (andkochergin) [reply spec/features/search_spec.rb]: done

File: spec/features/search_spec.rb (hunk 2 above)
Comment (andkochergin) [reply spec/features/search_spec.rb]: done

File: spec/features/search_spec.rb (hunk 3 above)
Comment (andkochergin) [reply spec/features/search_spec.rb]: done

File: spec/page_objects/pages/home_page.rb (hunk 5 above)
Comment (andkochergin) [reply spec/page_objects/pages/home_page.rb]: done

File: spec/features/search_spec.rb [hunk 6]
```diff
@@ -1,15 +1,27 @@
...
+  let(:search_form) { ::PageObjects::Forms::SearchForm.new }
+
   it 'is iPhone X released in 2017' do
     visit '/'
-    find(:xpath, '//input[@class="fast-search__input"]').set "iPhone X"
-    within_frame find('iframe.modal-iframe') do
-      first(:xpath,'//a[contains(text(),"iPhone X")]').click
-    end
+    search_form.find_and_open_iphone_x
     expect(self).to have_xpath('//table[@class="product-specs__table"] '\
```
Comment (yahorzhylinski) [inline spec/features/search_spec.rb]: what about to wrap it into product page?

**Pull request merged**